### PR TITLE
50762_50548task_firstcommit

### PR DESCRIPTION
**Pull request by dina28**

### CODE REVIEW

File: fels_100/resultViewController.m [hunk 1]
```diff
@@ -66,4 +67,15 @@ - (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPa
     return  60.0f;
 }
 
+- (IBAction)btnDoneAction:(id)sender {
+    categoriesViewController *categoriesviewcontroller= [[categoriesViewController alloc]init];
```
Comment (TequilarTomsk) [inline fels_100/resultViewController.m]: I dont understand why you init an controller here?

File: fels_100/testViewController.m [hunk 2]
```diff
@@ -92,6 +96,25 @@ - (IBAction)selectOptionAction:(id)sender {
...
 
+- (IBAction)btnSpechAction:(id)sender {
+    AVSpeechSynthesizer *synthesizer = [[AVSpeechSynthesizer alloc]init];
+    AVSpeechUtterance *utterance = [AVSpeechUtterance speechUtteranceWithString:self.questionContent.text];
+    [utterance setRate:0.2f];
+    [synthesizer speakUtterance:utterance];
+}
+
+- (IBAction)btnBackAction:(id)sender {
+    categoriesViewController *categoriesviewcontroller = [[categoriesViewController alloc]init];
```
Comment (TequilarTomsk) [inline fels_100/testViewController.m]: Same as above

File: fels_100/categoriesViewController.m [hunk 3]
```diff
@@ -100,8 +103,18 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
...
         testViewController.categoryTypeName = [ theDic objectForKey:@"name"];
-       // NSLog(@"%@",[ theDic objectForKey:@"id"]);
     }
 }
 
+- (IBAction)btnBackAction:(id)sender {
+    ProfileController *profilecontroller = [[ProfileController alloc]init];
+    UIViewController *sourceViewController = (UIViewController *)self;
+    UIViewController *destinationViewController = profilecontroller;
+    for (UIViewController* controller in sourceViewController.navigationController.viewControllers) {
```
Comment (nguyenvandungb) [inline fels_100/categoriesViewController.m]: Do not need init ProfileController like this for purpose check type of controller
You can check like this "[ProfileController class]"

File: fels_100/categoriesViewController.m [hunk 4]
```diff
@@ -97,10 +97,20 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
...
+        NSDictionary *theDic = [categoriesName objectAtIndex: indexPath.row ];
         testViewController.categoryType = [ theDic objectForKey:@"id"];
         testViewController.categoryTypeName = [ theDic objectForKey:@"name"];
-       // NSLog(@"%@",[ theDic objectForKey:@"id"]);
+    }
+}
+
+- (IBAction)btnBackAction:(id)sender {
+    UIViewController *sourceViewController = (UIViewController *)self;
+    UIViewController *destinationViewController = [ProfileController class];
```
Comment (TequilarTomsk) [inline fels_100/categoriesViewController.m]: just put the code : [ProfileController class] in condition below

File: fels_100/resultViewController.m [hunk 5]
```diff
@@ -66,4 +67,14 @@ - (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPa
     return  60.0f;
 }
 
+- (IBAction)btnDoneAction:(id)sender {
+    UIViewController *sourceViewController = (UIViewController *)self;
+    UIViewController *destinationViewController = [categoriesViewController class];
```
Comment (TequilarTomsk) [inline fels_100/resultViewController.m]: same as above

**Pull request merged**